helios-client/src/test/java/com/spotify/helios/client/AuthenticatingHttpConnectorTest.java

```diff
@@ -139,6 +140,20 @@ protected boolean matchesSafely(final URI item) {
     };
   }
 
+  private CustomTypeSafeMatcher<Map<String, List<String>>> hasKeys(final List<String> keys) {
```

Comment:
for maximum flexibility, this could take a `Matcher<Iterable<String>>` (or some other collection type) which would allow you to do `argThat(hasKeys(contains("Authorization"))` at the call site, to re-use the existing matcher. If hamcrest had a matcher that let you match on a `map.keys()` then you wouldn't need this at all even